src/test/java/ch/uzh/ifi/hase/soprafs22/service/UserServiceTest.java

```diff
@@ -165,9 +263,413 @@ void updateUser_Conflict(){ //Test what happens when user tries to change userna
+        Mockito.when(userRepository.findById(id)).thenReturn(testUser);
+        //Game added to user
+        userService.addGame(id, testGame);
+        assertEquals(false, userService.userHasNoActiveGame(id));
+    }
+
+    @Test
+    void assignWhiteCards_emptyList(){
+        long id = testUser.getId();
+        List<WhiteCard> emptyWhiteCardList = new ArrayList<>();
+
+        Mockito.when(userRepository.findById(id)).thenReturn(testUser);
+
+        //OverWrite testUsers WhiteCards with an EmptyList of White Cards
+        userService.assignWhiteCards(testUser.getId(), emptyWhiteCardList);
+
+        assertEquals(emptyWhiteCardList, testUser.getUserWhiteCards());
+    }
+
+    @Test
+    void assignWhiteCards_nonEmptyList(){
+        long id = testUser.getId();
+        //Creating white Card list with 2 whiteCards
+        WhiteCard testWhiteCard1 = new WhiteCard();
+        testWhiteCard1.setId(2L);
+        testWhiteCard1.setText("CardText2");
+        WhiteCard testWhiteCard2 = new WhiteCard();
+        testWhiteCard1.setId(3L);
+        testWhiteCard1.setText("CardText3");
+
+        List<WhiteCard> WhiteCardList = new ArrayList<>();
+        WhiteCardList.add(testWhiteCard1);
+        WhiteCardList.add(testWhiteCard2);
+
+        Mockito.when(userRepository.findById(id)).thenReturn(testUser);
+
+        //OverWrite testUsers WhiteCards with an EmptyList of White Cards
+        userService.assignWhiteCards(testUser.getId(), WhiteCardList);
+
+        assertEquals(WhiteCardList, testUser.getUserWhiteCards());
+    }
+
+    @Test
+    void createMatch_success(){
+        Match isMatch = userService.createMatch(testUser, otherUser);
+
+        //Comparing the First and second object of the pair of the matches. IF they are the same
+        //Then the match contains the same pairs
+        //Couldn't find a way to compare matches otherwise as the match is initialized in the method
+        assertEquals(testUser , isMatch.getUserPair().getObj1());
+        assertEquals(otherUser, isMatch.getUserPair().getObj2());
+    }
+
+    @Test
+    void deleteGameIfEmpty_emptyGame(){
+      Game emptyGame = new Game();
+      testUser.setActiveGame(emptyGame);
+      //This moves the emptyGame to past Games
+      testUser.flushGameToPastGames();
+      userService.deleteGameIfEmpty(testUser, emptyGame);
+      //getPastGames should now be empty as the emptygame got deleted
+      assertEquals(Collections.emptyList(), testUser.getPastGames());
+    }
+
+    @Test
+    void deleteGameIfEmpty_nonEmptyGame(){
+        Play play = new Play();
+        Game nonEmptyGame = new Game();
+        nonEmptyGame.enqueuePlay(play);
+        testUser.setActiveGame(nonEmptyGame);
+        //This moves the emptyGame to past Games
+        testUser.flushGameToPastGames();
+        userService.deleteGameIfEmpty(testUser, nonEmptyGame);
+        //Should now not return empty list for past games
+        assertNotEquals(Collections.emptyList(), testUser.getPastGames());
```

Comment:
I like this Collections.emptyList. is there a reason for not using it for all tests similar to this?

Reply:
@JoMuelle could you please answer as it is your code..